graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/nodes/promise/ImportCallNode.java

```diff
@@ -283,7 +332,56 @@ private boolean shouldCatch(Throwable exception) {
             }
         }
 
-        JavaScriptRootNode root = context.isOptionTopLevelAwait() ? new TopLevelAwaitImportModuleDynamicallyRootNode() : new ImportModuleDynamicallyRootNode();
+        class ModuleBlockRootNode extends JavaScriptRootNode {
+
+            @Child protected JavaScriptNode argumentNode = AccessIndexedArgumentNode.create(0);
+
+            @Override
+            public Object execute(VirtualFrame frame) {
+                @SuppressWarnings("unchecked")
```

Comment:
is this type cast always working? I'd add an assertion

Reply:
Assertions added.